sources/tools/Stride.Importer.3D/MeshConverter.cs

```diff
@@ -67,6 +75,47 @@ private void ResetConversionData()
             textureNameCount.Clear();
         }
 
+        private void DecideKeptMeshIndexFromOutput()
+        {
+            keptMeshIndex = -1;
+            keptMeshNameHint = null;
+
+            try
+            {
+                var name = System.IO.Path.GetFileNameWithoutExtension(vfsOutputFilename);
+                if (string.IsNullOrEmpty(name))
+                    return;
+
+                // "(All)" => all meshes
+                if (name.EndsWith(" (All)", StringComparison.OrdinalIgnoreCase))
+                    return;
+
+                // "(Mesh X)" => keep mesh X-1 (legacy)
+                const string tag = " (Mesh ";
+                if (name.EndsWith(")", StringComparison.Ordinal))
+                {
+                    var start = name.LastIndexOf(tag, StringComparison.OrdinalIgnoreCase);
+                    if (start >= 0)
+                    {
+                        var numStr = name.Substring(start + tag.Length, name.Length - (start + tag.Length) - 1);
+                        if (int.TryParse(numStr, out var oneBased) && oneBased >= 1)
+                        {
+                            keptMeshIndex = oneBased - 1;
+                            return;
+                        }
+                    }
+                }
+
+                // Otherwise: remember the full output name; we will try to match a mesh by name later
+                keptMeshNameHint = name;
+            }
+            catch
+            {
+                keptMeshIndex = -1;
+                keptMeshNameHint = null;
+            }
+        }
+
```

Comment:
I can tell that this is not going to work, far too fragile, and it doesn't, as soon as a user changes the name of a model, the model changes to another or to the whole model instead.

You can add more properties to the `ModelAsset` if needed.